src/spec-configuration/containerFeaturesOCI.ts

```diff
@@ -0,0 +1,197 @@
+import * as path from 'path';
+import * as tar from 'tar';
+import { request } from '../spec-utils/httpRequest';
+import { Log, LogLevel } from '../spec-utils/log';
+import { mkdirpLocal, writeLocalFile } from '../spec-utils/pfs';
+import { FeatureSet } from './containerFeaturesConfiguration';
+
+export interface OCIFeatureRef {
+    id: string;
+    version: string;
+    featureName: string;
+    owner: string;
+    namespace: string;
+    registry: string;
+}
+
+export interface OCIManifest {
+    schemaVersion: number;
+    mediaType: string;
+    config: {
+        mediaType: string;
+        digest: string;
+        size: number;
+    };
+    layers: [
+        {
+            mediaType: string;
+            digest: string;
+            size: number;
+            annotations: {
+                'org.opencontainers.image.ref.name': string;
+                'org.opencontainers.image.title': string;
+            };
+        }
+    ];
+}
+
+export function getOCIFeatureSet(output: Log, identifier: string, options: boolean | string | Record<string, boolean | string | undefined>, manifest: OCIManifest): FeatureSet {
+
+    const featureRef = getFeatureRef(output, identifier);
+
+    const feat = {
+        id: featureRef.id,
+        name: featureRef.id,
+        included: true,
+        value: options
+    };
+
+    let featureSet: FeatureSet = {
+        sourceInformation: {
+            type: 'oci',
+            manifest: manifest,
+            featureRef: featureRef,
+
+        },
+        features: [feat],
+    };
+
+    return featureSet;
+}
+
+export function getFeatureRef(output: Log, identifier: string): OCIFeatureRef {
+
+    // ex: ghcr.io/codspace/features/ruby:1
+    const splitOnColon = identifier.split(':');
+    const id = splitOnColon[0];
+    const version = splitOnColon[1] ? splitOnColon[1] : '1';
```

Comment:
In the spec (and in our publishing tooling) we can emphasize the convention we always publish a `latest` tag (for tags that indicate a new latest breakfix version according to semver), and hardcode version-less features to `latest` instead of 1?

Reply:
Yep missed it.